### PR TITLE
Differential Drive Train

### PR DESCRIPTION
**Pull request by msoucy**
Creates a `Subsystem` that contains all the main functionality of a differential drive.

Definitely needs testing first.

### CODE REVIEW

File: core/src/main/java/com/chopshop166/chopshoplib/drive/DiffDriveSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+     * @param degrees The angle to turn by in degrees.
+     * @param speed   The speed to turn at in motor controller units.
+     * @return The command.
+     */
+    public CommandBase turnDegrees(final double degrees, final double speed) {
+        return functional("Turn Degrees", () -> {
+            resetGyro();
+        }, () -> {
+            double realSpeed = speed;
+            if (degrees < 0 && speed > 0) {
```
Comment (bot190) [inline core/src/main/java/com/chopshop166/chopshoplib/drive/DiffDriveSubsystem.java:204]: What if `degrees > 0`, but `speed < 0`? We probably want to make the signs match.

File: core/src/main/java/com/chopshop166/chopshoplib/drive/DiffDriveSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+            driveTrain.arcadeDrive(speed, 0);
+        }, (interrupted) -> {
+            driveTrain.stopMotor();
+        }, () -> {
+            return encoderAvg() >= distance;
+        });
+    }
+
+    /**
+     * Turn to a given relative angle at the given speed.
```
Comment (bot190) [inline core/src/main/java/com/chopshop166/chopshoplib/drive/DiffDriveSubsystem.java:193]: I think it's worth noting that this resets the gyro.

**Comment by bot190** [comment]
Looks good overall. This'll certainly be nice to have in the future.

**Review by bot190 — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/chopshop166/chopshoplib/drive/DiffDriveSubsystem.java (hunk 1 above)
Comment (msoucy) [reply core/src/main/java/com/chopshop166/chopshoplib/drive/DiffDriveSubsystem.java:204]: "If we're turning left but the speed will turn us to the right"

Maybe we should be comparing the signs of the two?

**Review by bot190 — APPROVED** [review]

**Review by jeffaburke — APPROVED** [review]

**Pull request merged**